File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/external_catalog_migration_result.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class ExternalCatalogMigrationResult(object):
    """
    The details of catalog migration operation
    """

    def __init__(self, **kwargs):
        """
        Initializes a new ExternalCatalogMigrationResult object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param total_external_catalogs:
            The value to assign to the total_external_catalogs property of this ExternalCatalogMigrationResult.
        :type total_external_catalogs: int

        :param new_design_external_catalog_count:
            The value to assign to the new_design_external_catalog_count property of this ExternalCatalogMigrationResult.
        :type new_design_external_catalog_count: int

        :param migrated_external_catalogs:
            The value to assign to the migrated_external_catalogs property of this ExternalCatalogMigrationResult.
        :type migrated_external_catalogs: list[oci.aidataplatform_dp.models.CatalogMigrationDetail]

        :param failed_migration_external_catalogs:
            The value to assign to the failed_migration_external_catalogs property of this ExternalCatalogMigrationResult.
        :type failed_migration_external_catalogs: list[oci.aidataplatform_dp.models.CatalogMigrationDetail]

        :param migration_failed_with_exception:
            The value to assign to the migration_failed_with_exception property of this ExternalCatalogMigrationResult.
        :type migration_failed_with_exception: int

        """
        self.swagger_types = {
            'total_external_catalogs': 'int',
            'new_design_external_catalog_count': 'int',
            'migrated_external_catalogs': 'list[CatalogMigrationDetail]',
            'failed_migration_external_catalogs': 'list[CatalogMigrationDetail]',
            'migration_failed_with_exception': 'int'
        }

        self.attribute_map = {
            'total_external_catalogs': 'totalExternalCatalogs',
            'new_design_external_catalog_count': 'newDesignExternalCatalogCount',
            'migrated_external_catalogs': 'migratedExternalCatalogs',
            'failed_migration_external_catalogs': 'failedMigrationExternalCatalogs',
            'migration_failed_with_exception': 'migrationFailedWithException'
        }

        self._total_external_catalogs = None
        self._new_design_external_catalog_count = None
        self._migrated_external_catalogs = None
        self._failed_migration_external_catalogs = None
        self._migration_failed_with_exception = None

    @property
    def total_external_catalogs(self):
        """
        **[Required]** Gets the total_external_catalogs of this ExternalCatalogMigrationResult.
        Total External catalog present


        :return: The total_external_catalogs of this ExternalCatalogMigrationResult.
        :rtype: int
        """
        return self._total_external_catalogs

    @total_external_catalogs.setter
    def total_external_catalogs(self, total_external_catalogs):
        """
        Sets the total_external_catalogs of this ExternalCatalogMigrationResult.
        Total External catalog present


        :param total_external_catalogs: The total_external_catalogs of this ExternalCatalogMigrationResult.
        :type: int
        """
        self._total_external_catalogs = total_external_catalogs

    @property
    def new_design_external_catalog_count(self):
        """
        **[Required]** Gets the new_design_external_catalog_count of this ExternalCatalogMigrationResult.
        No of external catalogs with new design


        :return: The new_design_external_catalog_count of this ExternalCatalogMigrationResult.
        :rtype: int
        """
        return self._new_design_external_catalog_count

    @new_design_external_catalog_count.setter
    def new_design_external_catalog_count(self, new_design_external_catalog_count):
        """
        Sets the new_design_external_catalog_count of this ExternalCatalogMigrationResult.
        No of external catalogs with new design


        :param new_design_external_catalog_count: The new_design_external_catalog_count of this ExternalCatalogMigrationResult.
        :type: int
        """
        self._new_design_external_catalog_count = new_design_external_catalog_count

    @property
    def migrated_external_catalogs(self):
        """
        Gets the migrated_external_catalogs of this ExternalCatalogMigrationResult.
        No of external catalogs migrated successfully


        :return: The migrated_external_catalogs of this ExternalCatalogMigrationResult.
        :rtype: list[oci.aidataplatform_dp.models.CatalogMigrationDetail]
        """
        return self._migrated_external_catalogs

    @migrated_external_catalogs.setter
    def migrated_external_catalogs(self, migrated_external_catalogs):
        """
        Sets the migrated_external_catalogs of this ExternalCatalogMigrationResult.
        No of external catalogs migrated successfully


        :param migrated_external_catalogs: The migrated_external_catalogs of this ExternalCatalogMigrationResult.
        :type: list[oci.aidataplatform_dp.models.CatalogMigrationDetail]
        """
        self._migrated_external_catalogs = migrated_external_catalogs

    @property
    def failed_migration_external_catalogs(self):
        """
        Gets the failed_migration_external_catalogs of this ExternalCatalogMigrationResult.
        No of external catalogs migration failed


        :return: The failed_migration_external_catalogs of this ExternalCatalogMigrationResult.
        :rtype: list[oci.aidataplatform_dp.models.CatalogMigrationDetail]
        """
        return self._failed_migration_external_catalogs

    @failed_migration_external_catalogs.setter
    def failed_migration_external_catalogs(self, failed_migration_external_catalogs):
        """
        Sets the failed_migration_external_catalogs of this ExternalCatalogMigrationResult.
        No of external catalogs migration failed


        :param failed_migration_external_catalogs: The failed_migration_external_catalogs of this ExternalCatalogMigrationResult.
        :type: list[oci.aidataplatform_dp.models.CatalogMigrationDetail]
        """
        self._failed_migration_external_catalogs = failed_migration_external_catalogs

    @property
    def migration_failed_with_exception(self):
        """
        Gets the migration_failed_with_exception of this ExternalCatalogMigrationResult.
        Count of Catalog Failed with exception before migration


        :return: The migration_failed_with_exception of this ExternalCatalogMigrationResult.
        :rtype: int
        """
        return self._migration_failed_with_exception

    @migration_failed_with_exception.setter
    def migration_failed_with_exception(self, migration_failed_with_exception):
        """
        Sets the migration_failed_with_exception of this ExternalCatalogMigrationResult.
        Count of Catalog Failed with exception before migration


        :param migration_failed_with_exception: The migration_failed_with_exception of this ExternalCatalogMigrationResult.
        :type: int
        """
        self._migration_failed_with_exception = migration_failed_with_exception

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other

File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/ExternalCatalogMigrationResult.java
// Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.

package com.oracle.aidataplatform.dp.model;



/**
 * The details of catalog migration operation
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=ExternalCatalogMigrationResult.Builder.class)

public final class ExternalCatalogMigrationResult  {
    @Deprecated
    @java.beans.ConstructorProperties({"totalExternalCatalogs", "newDesignExternalCatalogCount", "migratedExternalCatalogs", "failedMigrationExternalCatalogs", "migrationFailedWithException"})
    public ExternalCatalogMigrationResult(Integer totalExternalCatalogs, Integer newDesignExternalCatalogCount, java.util.List<CatalogMigrationDetail> migratedExternalCatalogs, java.util.List<CatalogMigrationDetail> failedMigrationExternalCatalogs, Integer migrationFailedWithException) {
        super();
        this.totalExternalCatalogs = totalExternalCatalogs;
        this.newDesignExternalCatalogCount = newDesignExternalCatalogCount;
        this.migratedExternalCatalogs = migratedExternalCatalogs;
        this.failedMigrationExternalCatalogs = failedMigrationExternalCatalogs;
        this.migrationFailedWithException = migrationFailedWithException;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Total External catalog present
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("totalExternalCatalogs")
private Integer totalExternalCatalogs;

        /**
         * Total External catalog present
         * @param totalExternalCatalogs the value to set
         * @return this builder
         **/
        

public Builder totalExternalCatalogs(Integer totalExternalCatalogs) {
    this.totalExternalCatalogs = totalExternalCatalogs;
    return this;
}
            /**
     * No of external catalogs with new design
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("newDesignExternalCatalogCount")
private Integer newDesignExternalCatalogCount;

        /**
         * No of external catalogs with new design
         * @param newDesignExternalCatalogCount the value to set
         * @return this builder
         **/
        

public Builder newDesignExternalCatalogCount(Integer newDesignExternalCatalogCount) {
    this.newDesignExternalCatalogCount = newDesignExternalCatalogCount;
    return this;
}
            /**
     * No of external catalogs migrated successfully
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("migratedExternalCatalogs")
private java.util.List<CatalogMigrationDetail> migratedExternalCatalogs;

        /**
         * No of external catalogs migrated successfully
         * @param migratedExternalCatalogs the value to set
         * @return this builder
         **/
        

public Builder migratedExternalCatalogs(java.util.List<CatalogMigrationDetail> migratedExternalCatalogs) {
    this.migratedExternalCatalogs = migratedExternalCatalogs;
    return this;
}
            /**
     * No of external catalogs migration failed
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("failedMigrationExternalCatalogs")
private java.util.List<CatalogMigrationDetail> failedMigrationExternalCatalogs;

        /**
         * No of external catalogs migration failed
         * @param failedMigrationExternalCatalogs the value to set
         * @return this builder
         **/
        

public Builder failedMigrationExternalCatalogs(java.util.List<CatalogMigrationDetail> failedMigrationExternalCatalogs) {
    this.failedMigrationExternalCatalogs = failedMigrationExternalCatalogs;
    return this;
}
            /**
     * Count of Catalog Failed with exception before migration
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("migrationFailedWithException")
private Integer migrationFailedWithException;

        /**
         * Count of Catalog Failed with exception before migration
         * @param migrationFailedWithException the value to set
         * @return this builder
         **/
        

public Builder migrationFailedWithException(Integer migrationFailedWithException) {
    this.migrationFailedWithException = migrationFailedWithException;
    return this;
}


        public ExternalCatalogMigrationResult build() {
            ExternalCatalogMigrationResult model = new ExternalCatalogMigrationResult(this.totalExternalCatalogs
                , this.newDesignExternalCatalogCount
                , this.migratedExternalCatalogs
                , this.failedMigrationExternalCatalogs
                , this.migrationFailedWithException);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(ExternalCatalogMigrationResult model) {
                this.totalExternalCatalogs(model.getTotalExternalCatalogs());
    this.newDesignExternalCatalogCount(model.getNewDesignExternalCatalogCount());
    this.migratedExternalCatalogs(model.getMigratedExternalCatalogs());
    this.failedMigrationExternalCatalogs(model.getFailedMigrationExternalCatalogs());
    this.migrationFailedWithException(model.getMigrationFailedWithException());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Total External catalog present
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("totalExternalCatalogs")
    private final Integer totalExternalCatalogs;

        /**
     * Total External catalog present
     * @return the value
     **/
    
    public Integer getTotalExternalCatalogs() {
        return totalExternalCatalogs;
    }


        /**
     * No of external catalogs with new design
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("newDesignExternalCatalogCount")
    private final Integer newDesignExternalCatalogCount;

        /**
     * No of external catalogs with new design
     * @return the value
     **/
    
    public Integer getNewDesignExternalCatalogCount() {
        return newDesignExternalCatalogCount;
    }


        /**
     * No of external catalogs migrated successfully
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("migratedExternalCatalogs")
    private final java.util.List<CatalogMigrationDetail> migratedExternalCatalogs;

        /**
     * No of external catalogs migrated successfully
     * @return the value
     **/
    
    public java.util.List<CatalogMigrationDetail> getMigratedExternalCatalogs() {
        return migratedExternalCatalogs;
    }


        /**
     * No of external catalogs migration failed
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("failedMigrationExternalCatalogs")
    private final java.util.List<CatalogMigrationDetail> failedMigrationExternalCatalogs;

        /**
     * No of external catalogs migration failed
     * @return the value
     **/
    
    public java.util.List<CatalogMigrationDetail> getFailedMigrationExternalCatalogs() {
        return failedMigrationExternalCatalogs;
    }


        /**
     * Count of Catalog Failed with exception before migration
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("migrationFailedWithException")
    private final Integer migrationFailedWithException;

        /**
     * Count of Catalog Failed with exception before migration
     * @return the value
     **/
    
    public Integer getMigrationFailedWithException() {
        return migrationFailedWithException;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("ExternalCatalogMigrationResult(");
        sb.append("totalExternalCatalogs=").append(String.valueOf(this.totalExternalCatalogs));
        sb.append(", newDesignExternalCatalogCount=").append(String.valueOf(this.newDesignExternalCatalogCount));
        sb.append(", migratedExternalCatalogs=").append(String.valueOf(this.migratedExternalCatalogs));
        sb.append(", failedMigrationExternalCatalogs=").append(String.valueOf(this.failedMigrationExternalCatalogs));
        sb.append(", migrationFailedWithException=").append(String.valueOf(this.migrationFailedWithException));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof ExternalCatalogMigrationResult)) {
            return false;
        }

        ExternalCatalogMigrationResult other = (ExternalCatalogMigrationResult) o;
        return java.util.Objects.equals(this.totalExternalCatalogs, other.totalExternalCatalogs) &&
            java.util.Objects.equals(this.newDesignExternalCatalogCount, other.newDesignExternalCatalogCount) &&
            java.util.Objects.equals(this.migratedExternalCatalogs, other.migratedExternalCatalogs) &&
            java.util.Objects.equals(this.failedMigrationExternalCatalogs, other.failedMigrationExternalCatalogs) &&
            java.util.Objects.equals(this.migrationFailedWithException, other.migrationFailedWithException);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.totalExternalCatalogs == null ? 43 : this.totalExternalCatalogs.hashCode());
        result = (result * PRIME) + (this.newDesignExternalCatalogCount == null ? 43 : this.newDesignExternalCatalogCount.hashCode());
        result = (result * PRIME) + (this.migratedExternalCatalogs == null ? 43 : this.migratedExternalCatalogs.hashCode());
        result = (result * PRIME) + (this.failedMigrationExternalCatalogs == null ? 43 : this.failedMigrationExternalCatalogs.hashCode());
        result = (result * PRIME) + (this.migrationFailedWithException == null ? 43 : this.migrationFailedWithException.hashCode());
        return result;
    }


}
